File: digital-assignment-1/clahe.m
% constrast-limited adaptive histogram equalization
clc
clear
close all

% reading the input image
I = imread('images\picture1.jpg');
M = rgb2lab(I);
H = histeq(I);
L = M(:,:,1)/100;
L = adapthisteq(L,'NumTiles',[8 8],'ClipLimit',0.005);
M(:,:,1) = L*100;
R = lab2rgb(M);

% image after histogram equalization
subplot(2,2,1); 
imshow(H); 
title('Histogram Equalized Image');

% histogram of equalized image
subplot(2,2,2); 
imhist(H); 
title('Histogram Equalization');

% image after clahe is applied
subplot(2,2,3);
imshow(R); 
title('CLAHE Image');

% histogram of clahe image
subplot(2,2,4);
imhist(R);
title('CLAHE Histogram');

% metric to compare the results
nI =niqe(I);
nR = niqe(R);
fprintf("Image score for orginal image: %0.2f.\n", nI)
fprintf("Image score for clahe image: %0.2f.\n", nR)

File: digital-assignment-2/clahe.py
import cv2
import numpy as np

# Load the image
img = cv2.imread('images/picture1.jpg')

# Convert the image to grayscale
gray_img = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)

# Apply Contrast-Limited Adaptive Histogram Equalization (CLAHE)
clahe = cv2.createCLAHE(clipLimit=2.0, tileGridSize=(8,8))
clahe_img = clahe.apply(gray_img)

# Display the results
cv2.imshow('Input Image', img)
cv2.imshow('Enhanced Image', clahe_img)
cv2.waitKey(0)
cv2.destroyAllWindows()
